### DISCUSSION TITLE
PowerShell cells no longer working...

### DISCUSSION THREAD

**Discussion by juancrl** [microsoft/vscode-jupyter#10432] [Questions and Answers]
Apologies for the ambiguities, I'm no expert in notebooks.
I had several *.ipynb files with a mix of PowerShell code and markdown. Working perfectly until last week. Visual Studio Code insiders, .NET interactive notebooks, Jupyter notebooks, several kernels, all updated to the latest version.
Then there was an update on Visual Studio Code Insiders version, which probably triggered some updates on these extensions.
Now all notebooks give errors. I can only select a Python Kernel... even if the cells are marked as PowerShell, they seem to be interpreted as Python code. No PowerShell kernel at all.  pwsh 7.2.4 running ok outside the notebooks.

I've decided to start fresh, uninstalled all extensions, VSCode, deleted all evident related files, rebooted, and installed all again. VSCode insiders, then Python, then PowerShell extension, then .NET Interactive which installs this Jupyter extension. 
But after this, same situation, and only Python kernel available in the picker.

Another symptom : ".NET Interactive : Create new blank notebook" display "Activate Extensions"... and does nothing.

What is the right procedure (documentation of) to install the simplest environment which can run PowerShell, markdown, and maybe cells with C# and Python ?

**Comment by DonJayamanne** [comment]
> Activate Extensions"... and does nothing.

please could you open your console window using the command `Toggle Developer Tools` and check out there are any errors logged in there. if yes, please paste the error here. most likely the . net extension has failed to load

**Reply by juancrl** [reply]
Thanks Don (and others)
Good point (dev tools), I didn't think of it. Anyway, I've completely reinstalled VSCODE and some extensions, and this seems back to work. 

Some doubts, anyway. Most of them related to the documentation...

1. I'm row running VSCODE instead of VSCODE-Insiders. I thought Insiders was mandatory. Insiders is not needed anymore ?
2. I deleted the VSCODE* folder to reinstall. It seems it's not enough, better run Unins000.exe from that folder. Is there some documentation on persistent config data, maybe on registry ?
3. To run PowerShell inside a *.pyinb file, is the .NET Kernel enough ? Or Python is mandatory ? I think the previous version needed Python with some params to shell pwsh.exe
4. Where is the list of Kernels stored ? I cannot find any of those kernel.json files anymore.

Documentation on all of these (and the object model, I'd like to clean cells, or specify their width, by code) will be highly welcome.
Thanks in advance !

**Comment by DonJayamanne** [comment]
1 & 2 is best asked on vscode repo. 
3 can be answered by the . net interactive notebook extension repo.
4 - this depends on the kernels. if these are jupyter kernels , is stored in the standard location created by Jupyter notebooks application

**Reply by juancrl** [reply]
Thanks, I'll ask there, you're right.
As for 4, do you have any hint ? I've found %userprofile%\.vscode\extensions\mstoolsai.jupyter-2022.5.1001601848\temp\jupyter\kernels folder, but it is empty. No other 'kernels' folder is under my account profile.
